From 69dec66704ba043581e0e85b40cd3f824bc0fc9f Mon Sep 17 00:00:00 2001
From: Wang Zaikuo <zaikuo.wang@intel.com>
Date: Fri, 1 Jun 2018 15:56:50 +0800
Subject: [PATCH 1969/2367] media: intel-ipu4: psys: move resource tables to
 abi file

Message for Open Source:
Those tables are only used by ABI, so move to ABI specific file

Message for Internal:
[Issue/Feature] This is an improvement.
[Root Cause/Changes] Same as above.

Change-Id: I78ea5415938f277196ebd0e4340c916f44ce4a5d
Tracked-On: #JKRLCSN-256
Signed-off-by: Wang Zaikuo <zaikuo.wang@intel.com>
Signed-off-by: Meng Wei <wei.meng@intel.com>
---
 .../media/pci/intel/ipu4/ipu-platform-resources.h  |   5 -
 drivers/media/pci/intel/ipu4/ipu4-fw-resources.c   | 185 +++++++++++++++++++++
 drivers/media/pci/intel/ipu4/ipu4-resources.c      | 185 ---------------------
 3 files changed, 185 insertions(+), 190 deletions(-)

diff --git a/drivers/media/pci/intel/ipu4/ipu-platform-resources.h b/drivers/media/pci/intel/ipu4/ipu-platform-resources.h
index 68e26a4..3087f85 100644
--- a/drivers/media/pci/intel/ipu4/ipu-platform-resources.h
+++ b/drivers/media/pci/intel/ipu4/ipu-platform-resources.h
@@ -220,11 +220,6 @@ int ipu_psys_move_resources(const struct device *dev,
 void ipu_psys_free_resources(struct ipu_psys_resource_alloc *alloc,
 			     struct ipu_psys_resource_pool *pool);
 
-extern const u32 ipu_fw_psys_cell_types[];
-extern const u16 ipu_fw_num_dev_channels[];
-extern const u16 ipu_fw_psys_mem_size[];
-extern const enum ipu_mem_id ipu_fw_psys_cell_mem
-	[IPU_FW_PSYS_N_CELL_ID][IPU_FW_PSYS_N_MEM_TYPE_ID];
 extern const struct ipu_fw_resource_definitions *res_defs;
 
 #endif /* IPU_PLATFORM_RESOURCES_H */
diff --git a/drivers/media/pci/intel/ipu4/ipu4-fw-resources.c b/drivers/media/pci/intel/ipu4/ipu4-fw-resources.c
index 9177c3e..1a355f7 100644
--- a/drivers/media/pci/intel/ipu4/ipu4-fw-resources.c
+++ b/drivers/media/pci/intel/ipu4/ipu4-fw-resources.c
@@ -3,6 +3,191 @@
 
 #include "ipu-fw-psys.h"
 
+/* resources table */
+/*
+ * Cell types by cell IDs
+ */
+const u32 ipu_fw_psys_cell_types[IPU_FW_PSYS_N_CELL_ID] = {
+	IPU_FW_PSYS_SP_CTRL_TYPE_ID,
+	IPU_FW_PSYS_SP_SERVER_TYPE_ID,
+	IPU_FW_PSYS_SP_SERVER_TYPE_ID,
+	IPU_FW_PSYS_VP_TYPE_ID,
+	IPU_FW_PSYS_VP_TYPE_ID,
+	IPU_FW_PSYS_VP_TYPE_ID,
+	IPU_FW_PSYS_VP_TYPE_ID,
+	IPU_FW_PSYS_ACC_ISA_TYPE_ID,
+	IPU_FW_PSYS_ACC_PSA_TYPE_ID,
+	IPU_FW_PSYS_ACC_PSA_TYPE_ID,
+	IPU_FW_PSYS_ACC_PSA_TYPE_ID,
+	IPU_FW_PSYS_ACC_PSA_TYPE_ID,
+	IPU_FW_PSYS_ACC_PSA_TYPE_ID,
+	IPU_FW_PSYS_ACC_PSA_TYPE_ID,
+	IPU_FW_PSYS_ACC_OSA_TYPE_ID,
+	IPU_FW_PSYS_GDC_TYPE_ID,
+	IPU_FW_PSYS_GDC_TYPE_ID
+};
+
+const u16 ipu_fw_num_dev_channels[IPU_FW_PSYS_N_DEV_CHN_ID] = {
+	IPU_FW_PSYS_DEV_CHN_DMA_EXT0_MAX_SIZE,
+	IPU_FW_PSYS_DEV_CHN_GDC_MAX_SIZE,
+	IPU_FW_PSYS_DEV_CHN_DMA_EXT1_READ_MAX_SIZE,
+	IPU_FW_PSYS_DEV_CHN_DMA_EXT1_WRITE_MAX_SIZE,
+	IPU_FW_PSYS_DEV_CHN_DMA_INTERNAL_MAX_SIZE,
+	IPU_FW_PSYS_DEV_CHN_DMA_IPFD_MAX_SIZE,
+	IPU_FW_PSYS_DEV_CHN_DMA_ISA_MAX_SIZE,
+	IPU_FW_PSYS_DEV_CHN_DMA_FW_MAX_SIZE,
+#ifdef CONFIG_VIDEO_INTEL_IPU4P
+	IPU_FW_PSYS_DEV_CHN_DMA_CMPRS_MAX_SIZE
+#endif
+};
+
+const u16 ipu_fw_psys_mem_size[IPU_FW_PSYS_N_MEM_ID] = {
+	IPU_FW_PSYS_VMEM0_MAX_SIZE,
+	IPU_FW_PSYS_VMEM1_MAX_SIZE,
+	IPU_FW_PSYS_VMEM2_MAX_SIZE,
+	IPU_FW_PSYS_VMEM3_MAX_SIZE,
+	IPU_FW_PSYS_VMEM4_MAX_SIZE,
+	IPU_FW_PSYS_BAMEM0_MAX_SIZE,
+	IPU_FW_PSYS_BAMEM1_MAX_SIZE,
+	IPU_FW_PSYS_BAMEM2_MAX_SIZE,
+	IPU_FW_PSYS_BAMEM3_MAX_SIZE,
+	IPU_FW_PSYS_DMEM0_MAX_SIZE,
+	IPU_FW_PSYS_DMEM1_MAX_SIZE,
+	IPU_FW_PSYS_DMEM2_MAX_SIZE,
+	IPU_FW_PSYS_DMEM3_MAX_SIZE,
+	IPU_FW_PSYS_DMEM4_MAX_SIZE,
+	IPU_FW_PSYS_DMEM5_MAX_SIZE,
+	IPU_FW_PSYS_DMEM6_MAX_SIZE,
+	IPU_FW_PSYS_DMEM7_MAX_SIZE,
+	IPU_FW_PSYS_PMEM0_MAX_SIZE,
+	IPU_FW_PSYS_PMEM1_MAX_SIZE,
+	IPU_FW_PSYS_PMEM2_MAX_SIZE,
+	IPU_FW_PSYS_PMEM3_MAX_SIZE
+};
+
+const enum ipu_mem_id
+ipu_fw_psys_cell_mem[IPU_FW_PSYS_N_CELL_ID][IPU_FW_PSYS_N_MEM_TYPE_ID] = {
+	{
+	 IPU_FW_PSYS_N_MEM_ID,
+	 IPU_FW_PSYS_DMEM0_ID,
+	 IPU_FW_PSYS_N_MEM_ID,
+	 IPU_FW_PSYS_N_MEM_ID,
+	 IPU_FW_PSYS_N_MEM_ID
+	},
+	{
+	 IPU_FW_PSYS_N_MEM_ID,
+	 IPU_FW_PSYS_DMEM1_ID,
+	 IPU_FW_PSYS_N_MEM_ID,
+	 IPU_FW_PSYS_N_MEM_ID,
+	 IPU_FW_PSYS_N_MEM_ID
+	},
+	{
+	 IPU_FW_PSYS_N_MEM_ID,
+	 IPU_FW_PSYS_DMEM2_ID,
+	 IPU_FW_PSYS_N_MEM_ID,
+	 IPU_FW_PSYS_N_MEM_ID,
+	 IPU_FW_PSYS_N_MEM_ID
+	},
+	{
+	 IPU_FW_PSYS_VMEM4_ID,
+	 IPU_FW_PSYS_DMEM4_ID,
+	 IPU_FW_PSYS_VMEM0_ID,
+	 IPU_FW_PSYS_BAMEM0_ID,
+	 IPU_FW_PSYS_PMEM0_ID
+	},
+	{
+	 IPU_FW_PSYS_VMEM4_ID,
+	 IPU_FW_PSYS_DMEM5_ID,
+	 IPU_FW_PSYS_VMEM1_ID,
+	 IPU_FW_PSYS_BAMEM1_ID,
+	 IPU_FW_PSYS_PMEM1_ID
+	},
+	{
+	 IPU_FW_PSYS_VMEM4_ID,
+	 IPU_FW_PSYS_DMEM6_ID,
+	 IPU_FW_PSYS_VMEM2_ID,
+	 IPU_FW_PSYS_BAMEM2_ID,
+	 IPU_FW_PSYS_PMEM2_ID,
+	},
+	{
+	 IPU_FW_PSYS_VMEM4_ID,
+	 IPU_FW_PSYS_DMEM7_ID,
+	 IPU_FW_PSYS_VMEM3_ID,
+	 IPU_FW_PSYS_BAMEM3_ID,
+	 IPU_FW_PSYS_PMEM3_ID,
+	},
+	{
+	 IPU_FW_PSYS_N_MEM_ID,
+	 IPU_FW_PSYS_N_MEM_ID,
+	 IPU_FW_PSYS_N_MEM_ID,
+	 IPU_FW_PSYS_N_MEM_ID,
+	 IPU_FW_PSYS_N_MEM_ID
+	},
+	{
+	 IPU_FW_PSYS_N_MEM_ID,
+	 IPU_FW_PSYS_N_MEM_ID,
+	 IPU_FW_PSYS_N_MEM_ID,
+	 IPU_FW_PSYS_N_MEM_ID,
+	 IPU_FW_PSYS_N_MEM_ID
+	},
+	{
+	 IPU_FW_PSYS_N_MEM_ID,
+	 IPU_FW_PSYS_N_MEM_ID,
+	 IPU_FW_PSYS_N_MEM_ID,
+	 IPU_FW_PSYS_N_MEM_ID,
+	 IPU_FW_PSYS_N_MEM_ID
+	},
+	{
+	 IPU_FW_PSYS_N_MEM_ID,
+	 IPU_FW_PSYS_N_MEM_ID,
+	 IPU_FW_PSYS_N_MEM_ID,
+	 IPU_FW_PSYS_N_MEM_ID,
+	 IPU_FW_PSYS_N_MEM_ID
+	},
+	{
+	 IPU_FW_PSYS_N_MEM_ID,
+	 IPU_FW_PSYS_N_MEM_ID,
+	 IPU_FW_PSYS_N_MEM_ID,
+	 IPU_FW_PSYS_N_MEM_ID,
+	 IPU_FW_PSYS_N_MEM_ID
+	},
+	{
+	 IPU_FW_PSYS_N_MEM_ID,
+	 IPU_FW_PSYS_N_MEM_ID,
+	 IPU_FW_PSYS_N_MEM_ID,
+	 IPU_FW_PSYS_N_MEM_ID,
+	 IPU_FW_PSYS_N_MEM_ID
+	},
+	{
+	 IPU_FW_PSYS_N_MEM_ID,
+	 IPU_FW_PSYS_N_MEM_ID,
+	 IPU_FW_PSYS_N_MEM_ID,
+	 IPU_FW_PSYS_N_MEM_ID,
+	 IPU_FW_PSYS_N_MEM_ID
+	},
+	{
+	 IPU_FW_PSYS_N_MEM_ID,
+	 IPU_FW_PSYS_N_MEM_ID,
+	 IPU_FW_PSYS_N_MEM_ID,
+	 IPU_FW_PSYS_N_MEM_ID,
+	 IPU_FW_PSYS_N_MEM_ID
+	},
+	{
+	 IPU_FW_PSYS_N_MEM_ID,
+	 IPU_FW_PSYS_N_MEM_ID,
+	 IPU_FW_PSYS_N_MEM_ID,
+	 IPU_FW_PSYS_N_MEM_ID,
+	 IPU_FW_PSYS_N_MEM_ID
+	},
+	{
+	 IPU_FW_PSYS_N_MEM_ID,
+	 IPU_FW_PSYS_N_MEM_ID,
+	 IPU_FW_PSYS_N_MEM_ID,
+	 IPU_FW_PSYS_N_MEM_ID,
+	 IPU_FW_PSYS_N_MEM_ID
+	}
+};
+
 static const struct ipu_fw_resource_definitions default_defs = {
 	.cells = ipu_fw_psys_cell_types,
 	.num_cells = IPU_FW_PSYS_N_CELL_ID,
diff --git a/drivers/media/pci/intel/ipu4/ipu4-resources.c b/drivers/media/pci/intel/ipu4/ipu4-resources.c
index fba8c65..4f05fa0 100644
--- a/drivers/media/pci/intel/ipu4/ipu4-resources.c
+++ b/drivers/media/pci/intel/ipu4/ipu4-resources.c
@@ -12,191 +12,6 @@
 #include "ipu-fw-psys.h"
 #include "ipu-psys.h"
 
-/* resources table */
-/*
- * Cell types by cell IDs
- */
-const u32 ipu_fw_psys_cell_types[IPU_FW_PSYS_N_CELL_ID] = {
-	IPU_FW_PSYS_SP_CTRL_TYPE_ID,
-	IPU_FW_PSYS_SP_SERVER_TYPE_ID,
-	IPU_FW_PSYS_SP_SERVER_TYPE_ID,
-	IPU_FW_PSYS_VP_TYPE_ID,
-	IPU_FW_PSYS_VP_TYPE_ID,
-	IPU_FW_PSYS_VP_TYPE_ID,
-	IPU_FW_PSYS_VP_TYPE_ID,
-	IPU_FW_PSYS_ACC_ISA_TYPE_ID,
-	IPU_FW_PSYS_ACC_PSA_TYPE_ID,
-	IPU_FW_PSYS_ACC_PSA_TYPE_ID,
-	IPU_FW_PSYS_ACC_PSA_TYPE_ID,
-	IPU_FW_PSYS_ACC_PSA_TYPE_ID,
-	IPU_FW_PSYS_ACC_PSA_TYPE_ID,
-	IPU_FW_PSYS_ACC_PSA_TYPE_ID,
-	IPU_FW_PSYS_ACC_OSA_TYPE_ID,
-	IPU_FW_PSYS_GDC_TYPE_ID,
-	IPU_FW_PSYS_GDC_TYPE_ID
-};
-
-const u16 ipu_fw_num_dev_channels[IPU_FW_PSYS_N_DEV_CHN_ID] = {
-	IPU_FW_PSYS_DEV_CHN_DMA_EXT0_MAX_SIZE,
-	IPU_FW_PSYS_DEV_CHN_GDC_MAX_SIZE,
-	IPU_FW_PSYS_DEV_CHN_DMA_EXT1_READ_MAX_SIZE,
-	IPU_FW_PSYS_DEV_CHN_DMA_EXT1_WRITE_MAX_SIZE,
-	IPU_FW_PSYS_DEV_CHN_DMA_INTERNAL_MAX_SIZE,
-	IPU_FW_PSYS_DEV_CHN_DMA_IPFD_MAX_SIZE,
-	IPU_FW_PSYS_DEV_CHN_DMA_ISA_MAX_SIZE,
-	IPU_FW_PSYS_DEV_CHN_DMA_FW_MAX_SIZE,
-#ifdef CONFIG_VIDEO_INTEL_IPU4P
-	IPU_FW_PSYS_DEV_CHN_DMA_CMPRS_MAX_SIZE
-#endif
-};
-
-const u16 ipu_fw_psys_mem_size[IPU_FW_PSYS_N_MEM_ID] = {
-	IPU_FW_PSYS_VMEM0_MAX_SIZE,
-	IPU_FW_PSYS_VMEM1_MAX_SIZE,
-	IPU_FW_PSYS_VMEM2_MAX_SIZE,
-	IPU_FW_PSYS_VMEM3_MAX_SIZE,
-	IPU_FW_PSYS_VMEM4_MAX_SIZE,
-	IPU_FW_PSYS_BAMEM0_MAX_SIZE,
-	IPU_FW_PSYS_BAMEM1_MAX_SIZE,
-	IPU_FW_PSYS_BAMEM2_MAX_SIZE,
-	IPU_FW_PSYS_BAMEM3_MAX_SIZE,
-	IPU_FW_PSYS_DMEM0_MAX_SIZE,
-	IPU_FW_PSYS_DMEM1_MAX_SIZE,
-	IPU_FW_PSYS_DMEM2_MAX_SIZE,
-	IPU_FW_PSYS_DMEM3_MAX_SIZE,
-	IPU_FW_PSYS_DMEM4_MAX_SIZE,
-	IPU_FW_PSYS_DMEM5_MAX_SIZE,
-	IPU_FW_PSYS_DMEM6_MAX_SIZE,
-	IPU_FW_PSYS_DMEM7_MAX_SIZE,
-	IPU_FW_PSYS_PMEM0_MAX_SIZE,
-	IPU_FW_PSYS_PMEM1_MAX_SIZE,
-	IPU_FW_PSYS_PMEM2_MAX_SIZE,
-	IPU_FW_PSYS_PMEM3_MAX_SIZE
-};
-
-const enum ipu_mem_id
-ipu_fw_psys_cell_mem[IPU_FW_PSYS_N_CELL_ID][IPU_FW_PSYS_N_MEM_TYPE_ID] = {
-	{
-	 IPU_FW_PSYS_N_MEM_ID,
-	 IPU_FW_PSYS_DMEM0_ID,
-	 IPU_FW_PSYS_N_MEM_ID,
-	 IPU_FW_PSYS_N_MEM_ID,
-	 IPU_FW_PSYS_N_MEM_ID
-	},
-	{
-	 IPU_FW_PSYS_N_MEM_ID,
-	 IPU_FW_PSYS_DMEM1_ID,
-	 IPU_FW_PSYS_N_MEM_ID,
-	 IPU_FW_PSYS_N_MEM_ID,
-	 IPU_FW_PSYS_N_MEM_ID
-	},
-	{
-	 IPU_FW_PSYS_N_MEM_ID,
-	 IPU_FW_PSYS_DMEM2_ID,
-	 IPU_FW_PSYS_N_MEM_ID,
-	 IPU_FW_PSYS_N_MEM_ID,
-	 IPU_FW_PSYS_N_MEM_ID
-	},
-	{
-	 IPU_FW_PSYS_VMEM4_ID,
-	 IPU_FW_PSYS_DMEM4_ID,
-	 IPU_FW_PSYS_VMEM0_ID,
-	 IPU_FW_PSYS_BAMEM0_ID,
-	 IPU_FW_PSYS_PMEM0_ID
-	},
-	{
-	 IPU_FW_PSYS_VMEM4_ID,
-	 IPU_FW_PSYS_DMEM5_ID,
-	 IPU_FW_PSYS_VMEM1_ID,
-	 IPU_FW_PSYS_BAMEM1_ID,
-	 IPU_FW_PSYS_PMEM1_ID
-	},
-	{
-	 IPU_FW_PSYS_VMEM4_ID,
-	 IPU_FW_PSYS_DMEM6_ID,
-	 IPU_FW_PSYS_VMEM2_ID,
-	 IPU_FW_PSYS_BAMEM2_ID,
-	 IPU_FW_PSYS_PMEM2_ID,
-	},
-	{
-	 IPU_FW_PSYS_VMEM4_ID,
-	 IPU_FW_PSYS_DMEM7_ID,
-	 IPU_FW_PSYS_VMEM3_ID,
-	 IPU_FW_PSYS_BAMEM3_ID,
-	 IPU_FW_PSYS_PMEM3_ID,
-	},
-	{
-	 IPU_FW_PSYS_N_MEM_ID,
-	 IPU_FW_PSYS_N_MEM_ID,
-	 IPU_FW_PSYS_N_MEM_ID,
-	 IPU_FW_PSYS_N_MEM_ID,
-	 IPU_FW_PSYS_N_MEM_ID
-	},
-	{
-	 IPU_FW_PSYS_N_MEM_ID,
-	 IPU_FW_PSYS_N_MEM_ID,
-	 IPU_FW_PSYS_N_MEM_ID,
-	 IPU_FW_PSYS_N_MEM_ID,
-	 IPU_FW_PSYS_N_MEM_ID
-	},
-	{
-	 IPU_FW_PSYS_N_MEM_ID,
-	 IPU_FW_PSYS_N_MEM_ID,
-	 IPU_FW_PSYS_N_MEM_ID,
-	 IPU_FW_PSYS_N_MEM_ID,
-	 IPU_FW_PSYS_N_MEM_ID
-	},
-	{
-	 IPU_FW_PSYS_N_MEM_ID,
-	 IPU_FW_PSYS_N_MEM_ID,
-	 IPU_FW_PSYS_N_MEM_ID,
-	 IPU_FW_PSYS_N_MEM_ID,
-	 IPU_FW_PSYS_N_MEM_ID
-	},
-	{
-	 IPU_FW_PSYS_N_MEM_ID,
-	 IPU_FW_PSYS_N_MEM_ID,
-	 IPU_FW_PSYS_N_MEM_ID,
-	 IPU_FW_PSYS_N_MEM_ID,
-	 IPU_FW_PSYS_N_MEM_ID
-	},
-	{
-	 IPU_FW_PSYS_N_MEM_ID,
-	 IPU_FW_PSYS_N_MEM_ID,
-	 IPU_FW_PSYS_N_MEM_ID,
-	 IPU_FW_PSYS_N_MEM_ID,
-	 IPU_FW_PSYS_N_MEM_ID
-	},
-	{
-	 IPU_FW_PSYS_N_MEM_ID,
-	 IPU_FW_PSYS_N_MEM_ID,
-	 IPU_FW_PSYS_N_MEM_ID,
-	 IPU_FW_PSYS_N_MEM_ID,
-	 IPU_FW_PSYS_N_MEM_ID
-	},
-	{
-	 IPU_FW_PSYS_N_MEM_ID,
-	 IPU_FW_PSYS_N_MEM_ID,
-	 IPU_FW_PSYS_N_MEM_ID,
-	 IPU_FW_PSYS_N_MEM_ID,
-	 IPU_FW_PSYS_N_MEM_ID
-	},
-	{
-	 IPU_FW_PSYS_N_MEM_ID,
-	 IPU_FW_PSYS_N_MEM_ID,
-	 IPU_FW_PSYS_N_MEM_ID,
-	 IPU_FW_PSYS_N_MEM_ID,
-	 IPU_FW_PSYS_N_MEM_ID
-	},
-	{
-	 IPU_FW_PSYS_N_MEM_ID,
-	 IPU_FW_PSYS_N_MEM_ID,
-	 IPU_FW_PSYS_N_MEM_ID,
-	 IPU_FW_PSYS_N_MEM_ID,
-	 IPU_FW_PSYS_N_MEM_ID
-	}
-};
-
 static int ipu_resource_init(struct ipu_resource *res, u32 id, int elements)
 {
 	if (elements <= 0) {
-- 
2.7.4

